x-pack/libbeat/common/aws/credentials.go

```diff
@@ -185,7 +185,7 @@ func EnrichAWSConfigWithEndpoint(endpoint string, serviceName string, regionName
 
 		beatsConfig.EndpointResolverWithOptions = awssdk.EndpointResolverWithOptionsFunc(
 			func(service, region string, options ...interface{}) (awssdk.Endpoint, error) {
-				return awssdk.Endpoint{URL: eurl}, nil
+				return awssdk.Endpoint{URL: eurl, SigningRegion: region}, nil
```

Comment:
Should we add a check `endpoint != "amazonaws.com"` at line 171?

Reply:
if the workaround is to have a zero length endpoint basically we are bypassing the call to `EnrichAWSConfigWithEndpoint`

given so I'm going to push a commit removing all the calls to `EnrichAWSConfigWithEndpoint` and rely on every service internal endpoint resolver from the sdk.

the only scenario were we don't access aws endpoint should be in the `s3poller`, where we can have non-aws s3 compatible bucket. I will test what's needed there.

Reply:
it seems this can be fixed asap in integrations with removing the default value at https://github.com/elastic/integrations/blob/main/packages/aws/manifest.yml#L68

and releasing a new package version

Reply:
https://github.com/elastic/integrations/pull/4103